src/tests/unit/tests/background/stores/card-selection-store.test.ts

```diff
@@ -231,6 +231,23 @@ describe('CardSelectionStore Test', () => {
         createStoreForCardSelectionActions('collapseAllRules').testListenerToBeCalledOnce(initialState, expectedState);
     });
 
+    test('expandAllRules', () => {
+        initialState.rules['sampleRuleId1'].isExpanded = true;
+        initialState.rules['sampleRuleId1'].cards['sampleUid1'] = true;
+
+        expectedState.rules['sampleRuleId1'].isExpanded = true;
+        expectedState.rules['sampleRuleId1'].cards['sampleUid1'] = true;
+        expectedState.rules['sampleRuleId2'].isExpanded = true;
+
+        createStoreForCardSelectionActions('expandAllRules').testListenerToBeCalledOnce(initialState, expectedState);
+    });
+
+    test('toggleVisualHelper on - no card selection or rule expansion changes', () => {});
+
+    test('toggleVisualHelper off - cards deselected, no rule expansion changes', () => {});
+
+    test('select card, turn on visual helper', () => {});
```

Comment:
Wouldn't this just be another case of the "toggleCardSelection"? In the case where it is enabled, we also validate the visual helper is enabled.

Reply:
I favor testing only one thing per test function. Especially because the two effects are not necessarily related. However, since I removed the empty test function, it's not an issue for now.